test/integration/api/authentication.spec.js

```diff
@@ -15,6 +15,69 @@ const { encodeBase64 } = require('../../../src/util/encoding')
 
 const createOidcServer = require('../../../test/oauth/server')
 
+const OAUTH_TEST_CONFIGS = (() => {
+  try {
+    /**
+     * process.env.OAUTH_TEST_CONFIGS must be a JSON stringified object with the following format:
+     * {
+     *   "<provider>": {
+     *     "credentials": {
+     *       "username": ""
+     *       "password": ""
+     *     },
+     *     "ssoConnection": {
+     *       "clientId": "",
+     *       "clientSecret": ""
+     *     }
+     *   }
+     * }
+     *
+     * Currently, the following providers are tested: 'github' and 'facebook'
+     */
+
+    return JSON.parse(process.env.OAUTH_TEST_CONFIGS)
+  } catch (err) {
+    return {}
+  }
+})()
+
+const shouldExecuteBuiltInSSOTest = (() => {
+  return typeof !process.env.CIRCLECI ||
+    (
+      process.env.CIRCLECI &&
+        (process.env.CIRCLE_BRANCH !== 'dev' || !process.env.CIRCLE_BRANCH.startsWith('dependabot/'))
```

Comment:
Maybe the body of your commit should also appear here to explain why you exclude these specific branches.